### PR TITLE
docs(react-dialog): improves lazy loading content docs

### PR DESCRIPTION
**Pull request by bsunderhus**
## New Behavior

Improves documentation on the `No Focusable Element` scenario, describing the case of lazy loading content

image

## Related Issue(s)

- Fixes https://github.com/microsoft/fluentui/issues/28371

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28375/perf-test-react-components/Avatar-mount_base.html">590</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28375/perf-test-react-components/Avatar-mount.html">594</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28375/perf-test-react-components/Button-mount_base.html">292</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28375/perf-test-react-components/Button-mount.html">297</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28375/perf-test-react-components/Field-mount_base.html">1043</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28375/perf-test-react-components/Field-mount.html">1068</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28375/perf-test-react-components/FluentProvider-mount_base.html">651</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28375/perf-test-react-components/FluentProvider-mount.html">668</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28375/perf-test-react-components/FluentProviderWithTheme-mount_base.html">77</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28375/perf-test-react-components/FluentProviderWithTheme-mount.html">86</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28375/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">70</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28375/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">64</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28375/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">74</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28375/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">65</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28375/perf-test-react-components/InfoButton-mount_base.html">14</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28375/perf-test-react-components/InfoButton-mount.html">17</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28375/perf-test-react-components/MakeStyles-mount_base.html">846</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28375/perf-test-react-components/MakeStyles-mount.html">862</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28375/perf-test-react-components/Persona-mount_base.html">1659</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28375/perf-test-react-components/Persona-mount.html">1615</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28375/perf-test-react-components/SpinButton-mount_base.html">1306</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28375/perf-test-react-components/SpinButton-mount.html">1352</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `65.162 kB`<br />`17.952 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `210.395 kB`<br />`58.717 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `36.395 kB`<br />`11.996 kB` |
| <samp>react-dialog</samp> <br /> <abbr title='bundle-size/Dialog.fixture.js'>Dialog (including children components)</abbr>                                                    |  `92.138 kB`<br />`27.487 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `6.473 kB`<br />`2.196 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/7d519bdcaef0a3249b6eae721fea0e5f096f6d35'>7d519bdcaef0a3249b6eae721fea0e5f096f6d35</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

File: packages/react-components/react-dialog/stories/Dialog/DialogNoFocusableElement.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 A `Dialog` **should** always have at least one focusable element.
 Some accessibility issues might happen if no focusable element is provided, like this one caught in [Talkback](https://issuetracker.google.com/issues/243456562?pli=1).
 
-In the case no focusable element is present inside a `Dialog` the only method
-that would close the `Dialog` would be clicking on the `backdrop`
+In the case no focusable element is present inside a `Dialog` the only method that would close the `Dialog` would be clicking on the `backdrop`.
```
Comment (ling1726) [inline packages/react-components/react-dialog/stories/Dialog/DialogNoFocusableElement.md:4]:
```suggestion
In the case when there is no focusable element inside a `Dialog` the only way to close the `Dialog` would be clicking on the `backdrop`.
```

File: packages/react-components/react-dialog/stories/Dialog/DialogNoFocusableElement.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
...
+
+> A common scenario for no focusable elements on a dialog is lazy loading content, where the focusable elements are only added after the content is loaded. In that case, it is recommended to manually focus on the desired focusable element after the content is loaded.
```
Comment (ling1726) [inline packages/react-components/react-dialog/stories/Dialog/DialogNoFocusableElement.md:6]:
```suggestion
> A common scenario for no focusable elements on a dialog is lazy loaded content, where the content (with focusable elements) is added after the Dialog is mounted. In that case, it is recommended to manually focus on the desired focusable element after the content is loaded.
```

**Review by ling1726 — APPROVED** [review]

**Pull request merged**